Add Amazon Linux support

CW-B-W opened the pull request:
Tested on `Amazon Linux 2023` & `Amazon Linux 2`

angristan commented on wireguard-install.sh:
```diff
@@ -204,6 +206,19 @@ function installWireGuard() {
+		
+		# Enable WireGuard kernel module
+		echo "Enabling WireGuard kernel module..."
+		modprobe wireguard
```
This isn't done by default? 👀

angristan commented:
Thanks, can you update the readme as well?

CW-B-W commented:
> Thanks, can you update the readme as well?

Sure, updated.

CW-B-W committed:
feat: modify for ec2 installation

CW-B-W committed:
feat: add WireGuard Manager script

CW-B-W committed:
Merge upstream/master into ec2 branch

Resolved merge conflict in wireguard-install.sh by keeping upstream's installPackages function usage for better error handling.

CW-B-W committed:
Fix duplicate port question and update getHomeDirForClient function in wireguard-manager.sh

- Remove duplicate SERVER_PORT question that appeared twice
- Add default SERVER_PORT=51820 before port prompt (consistent with wireguard-install.sh)
- Update getHomeDirForClient to accept CLIENT_NAME parameter (fixes function signature mismatch)

CW-B-W committed:
Add curl back to package installations to preserve ec2 modifications

The ec2 commit dd5a6f5 added curl to all package installation commands
since the script uses curl on line 119 to get public IP. The upstream
version removed curl, causing the script to fail if curl isn't installed.
This preserves the ec2 modification while keeping upstream's installPackages
function for better error handling.

CW-B-W committed:
feat: enhance WireGuard management with conflict checks and config viewing

- Remove SERVER_PORT=51820 default, allow user to enter desired port
- Add IP and port conflict checks for existing WireGuard interfaces
- Add 'Show user configuration' option with normal/VLAN modes
- Normal mode: Shows full VPN configuration with DNS
- VLAN mode: Shows subnet-only config without DNS, adds PersistentKeepalive
- Add PersistentKeepalive = 25 to all client configurations
- Update both wireguard-install.sh and wireguard-manager.sh scripts

CW-B-W committed:
fix: add +x permissions

CW-B-W committed:
Add configurable MTU support with intelligent defaults

- Detect default NIC MTU and suggest WireGuard MTU as (NIC_MTU - 80)
- Allow user to configure MTU during interface creation
- Store WG_MTU in parameters for consistency
- Apply configured MTU to both server and client configurations
- Remove hardcoded MTU=1420 in favor of calculated defaults

CW-B-W committed:
refactor(wireguard-install): convert to pure installer, delegate management to wireguard-manager.sh
